centrallix/utility/obfuscate.c

```diff
@@ -1211,6 +1212,7 @@ obfObfuscateData(pObjData srcval, pObjData dstval, int data_type, char* attrname
 		    }
 		//m.WholePart = floor(dv/100.0);
 		//m.FractionPart = (dv - m.WholePart*100) * 100;
+		m.Value = dv * 100;
```

Comment:
The original code is pretty gnarly - you got it right as far as I can tell, but just make sure you run tests on the old + new versions to make sure you get the same results!